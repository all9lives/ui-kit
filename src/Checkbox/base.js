import React, { PureComponent } from 'react'
import { css } from 'styled-components'
import Unit from '../utils/Unit'
import Spacing from '../utils/Spacing'
import Text from '../Text'

const base = css`
  position: relative;
  margin-right: ${Unit.convert(Spacing.regular)};
`

export const getComponent = ({ View, Checkbox }) => class Component extends PureComponent {
  handleValueChange = (value) => {
    if (this.props.onValueChange) {
      this.props.onValueChange(value)
    }
  }
  handleLabelPress = () => {
    if (this.props.onValueChange) {
      this.props.onValueChange(!this.props.value)
    }
  }
  render () {
    const { value, label } = this.props
    return (
      <View align='middle'>
        <Checkbox value={value} onValueChange={this.handleValueChange} />
        <Text alignSelf='center' onPress={this.handleLabelPress}>{label}</Text>
      </View>
    )
  }
}

export default base
