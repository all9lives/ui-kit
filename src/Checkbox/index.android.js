import styled from 'styled-components/native'
import base, { getComponent } from './base'
import defaultProps from './defaultProps'
import View from '../View'
import Color from '../utils/Color'

export const StyledView = styled(View)`
  flex-direction: row;
`

export const getStyled = attrs => styled.Switch.attrs(attrs)`
  ${base}
`

export const Checkbox = getStyled({
  onTintColor: props => (props.value ? Color.primary_light : undefined),
  thumbTintColor: props => (props.value ? Color.primary : Color.gray)
})
Checkbox.defaultProps = defaultProps

export default getComponent({ View: StyledView, Checkbox })
