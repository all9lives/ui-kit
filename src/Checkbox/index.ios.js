import { StyledView, getStyled } from './index.android'
import { getComponent } from './base'
import defaultProps from './defaultProps'
import Color from '../utils/Color'

export const Checkbox = getStyled({
  onTintColor: Color.primary,
  thumbTintColor: Color.white,
  tintColor: Color.primary_light
})
Checkbox.defaultProps = defaultProps

export default getComponent({ View: StyledView, Checkbox })
