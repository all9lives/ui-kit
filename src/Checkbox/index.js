import styled from 'styled-components'
import base, { getComponent } from './base'
import defaultProps from './defaultProps'
import View from '../View'

export const Checkbox = styled.input.attrs({
  onChange: props => (e) => {
    if (props.onValueChange) {
      props.onValueChange(e.target.checked)
    }
  },
  checked: props => props.value,
  type: 'checkbox'
})`
  ${base}
`
Checkbox.defaultProps = defaultProps

export default getComponent({ View, Checkbox })
