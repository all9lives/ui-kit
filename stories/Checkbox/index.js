import React from 'react'
import createStory from '../../storybook/createStory'
import { CheckboxWrapper } from '../../storybook/createStory/InputWrapper'
import Checkbox from '../../src/Checkbox'

createStory('Checkbox', module)
  .add('with `label:Check me`', () => (
    <CheckboxWrapper>
      <Checkbox label='Check Me' />
    </CheckboxWrapper>
  ))
