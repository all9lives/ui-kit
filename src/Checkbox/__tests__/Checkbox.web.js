/* eslint-env jest */
import 'jest-styled-components'
import React from 'react'
import { shallow } from 'enzyme'
import Checkbox, { Checkbox as Input } from '../index.js'

describe('Checkbox.web', () => {
  it('renders a snapshot', () => {
    const wrapper = shallow(
      <Checkbox label='FOOBAR' />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `onValueChange`', () => {
    const wrapper = shallow(
      <Checkbox onValueChange={() => {}} />
    )
    wrapper.instance().handleValueChange(true)
    wrapper.update()
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot without `onValueChange`', () => {
    const wrapper = shallow(
      <Checkbox />
    )
    wrapper.instance().handleValueChange(true)
    wrapper.update()
    expect(wrapper).toMatchSnapshot()
  })

  it('triggers `handleLabelPress`', () => {
    let result = null
    const wrapper = shallow(
      <Checkbox value onValueChange={(e) => (result = e)} />
    )
    wrapper.instance().handleLabelPress()
    expect(result).toMatchSnapshot()
  })

  it('does not triggers `handleLabelPress`', () => {
    let result = null
    const wrapper = shallow(
      <Checkbox value />
    )
    wrapper.instance().handleLabelPress()
    expect(result).toMatchSnapshot()
  })
})

describe('Checkbox.Checkbox.web', () => {
  it('renders a snapshot', () => {
    const wrapper = shallow(
      <Input />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `onChange`', () => {
    const wrapper = shallow(
      <Input />
    )
    wrapper.find('input').simulate('change', { target: { value: true } })
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `value:true`', () => {
    const wrapper = shallow(
      <Input value />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with `onChange & onValueChange`', () => {
    const wrapper = shallow(
      <Input onValueChange={() => {}} />
    )
    wrapper.find('input').simulate('change', { target: { value: true } })
    expect(wrapper).toMatchSnapshot()
  })
})
