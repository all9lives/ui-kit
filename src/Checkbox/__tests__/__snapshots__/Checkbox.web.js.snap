// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox.Checkbox.web renders a snapshot 1`] = `
.c0 {
  position: relative;
  margin-right: 10px;
}

<input
  className="c0"
  onChange={[Function]}
  type="checkbox"
/>
`;

exports[`Checkbox.Checkbox.web renders a snapshot with \`onChange & onValueChange\` 1`] = `
.c0 {
  position: relative;
  margin-right: 10px;
}

<input
  className="c0"
  onChange={[Function]}
  type="checkbox"
/>
`;

exports[`Checkbox.Checkbox.web renders a snapshot with \`onChange\` 1`] = `
.c0 {
  position: relative;
  margin-right: 10px;
}

<input
  className="c0"
  onChange={[Function]}
  type="checkbox"
/>
`;

exports[`Checkbox.Checkbox.web renders a snapshot with \`value:true\` 1`] = `
.c0 {
  position: relative;
  margin-right: 10px;
}

<input
  checked={true}
  className="c0"
  onChange={[Function]}
  type="checkbox"
  value={true}
/>
`;

exports[`Checkbox.web does not triggers \`handleLabelPress\` 1`] = `null`;

exports[`Checkbox.web renders a snapshot 1`] = `
<Styled(View)
  align="middle"
  direction="column"
  margin={0}
  padding={0}
>
  <styled.input
    onValueChange={[Function]}
  />
  <Styled(Text)
    alignSelf="center"
    color="#191919"
    margin={0}
    onPress={[Function]}
    size={16}
  >
    FOOBAR
  </Styled(Text)>
</Styled(View)>
`;

exports[`Checkbox.web renders a snapshot with \`onValueChange\` 1`] = `
<Styled(View)
  align="middle"
  direction="column"
  margin={0}
  padding={0}
>
  <styled.input
    onValueChange={[Function]}
  />
  <Styled(Text)
    alignSelf="center"
    color="#191919"
    margin={0}
    onPress={[Function]}
    size={16}
  />
</Styled(View)>
`;

exports[`Checkbox.web renders a snapshot without \`onValueChange\` 1`] = `
<Styled(View)
  align="middle"
  direction="column"
  margin={0}
  padding={0}
>
  <styled.input
    onValueChange={[Function]}
  />
  <Styled(Text)
    alignSelf="center"
    color="#191919"
    margin={0}
    onPress={[Function]}
    size={16}
  />
</Styled(View)>
`;

exports[`Checkbox.web triggers \`handleLabelPress\` 1`] = `false`;
